crates/nu-protocol/src/errors/parse_warning.rs

```diff
@@ -14,12 +14,22 @@ pub enum ParseWarning {
         span: Span,
         url: String,
     },
+    #[error("Deprecated: {old}")]
+    #[diagnostic(help("{help}"))]
+    CustomDeprecatedWarning {
+        old: String,
+        help: String,
+        label: String,
+        #[label("{label}")]
+        span: Span,
+    },
```

Comment:
@sholderbach this is a bit ugly :sweat_smile: do you think this is reasonable, maybe there's something nicer we could do? the normal deprecation warning isn't really fit for this

Reply:
I'm not pinged here but maybe we should improve the warning behavior. I had some ideas how we could improve this

Reply:
Those partially templated errors are hair-tearing :D I don't think `ParseWarning` is particularly crowded yet. We can refine that if necessary later.